test/extensions/transport_sockets/alts/alts_integration_test.cc

```diff
@@ -87,6 +105,204 @@ class CapturingHandshakerService : public grpc::gcp::HandshakerService::Service
   size_t server_max_frame_size{0};
 };
 
+// FakeHandshakeService implements a fake handshaker service using a fake key
```

Comment:
Do we an an integration test where we concurrently try to make connections (say 2) and we have 2 workers? I think that would help give confidence in fixing the underlying issue (and not introducing another race)

Reply:
I have an integration test where I run many concurrent connections through an Envoy running on GCP using the real ALTS handshaker service - this is what I used to confirm the fix.

At present, it is not clear to me how to set up a full integration test in the Envoy repo that does many concurrent connections with a fake ALTS handshaker service. However, I set up unit tests at the TSI handshaker layer that do concurrent handshakes, and these also would have caught the original issue.